control_plane/src/pageserver.rs

```diff
@@ -127,10 +127,13 @@ impl PageServerNode {
         }
 
         // Apply the user-provided overrides
-        overrides.push(
-            toml_edit::ser::to_string_pretty(&conf)
-                .expect("we deserialized this from toml earlier"),
-        );
+        overrides.push({
+            let mut doc =
+                toml_edit::ser::to_document(&conf).expect("we deserialized this from toml earlier");
+            // `id` is written out to `identity.toml` instead of `pageserver.toml`
+            doc.remove("id").expect("it's part of the struct");
```

Comment:
Are we including id in the serialized output needlessly instead of `serde(skip_serializing)`... or does this serve some purpose I cannot see?

Reply:
This was intentional.

> Are we including id in the serialized output needlessly instead of #[serde(skip_serializing)]

The `NeonLocalInitPageserverConf` is documented to be

> /// The toml that can be passed to `neon_local init --config`.
> /// This is a subset of the `pageserver.toml` configuration.
> // TODO(christian): use pageserver_api::config::ConfigToml (PR #7656)

It is correct for it to include the `id` field.
We're not serializing it anywhere at this time, but still, `skip_serializing` feel like the wrong thing to do here.

The PR #7656 which I will tend do after we are done with `identity.toml` will hopefully clean up a bunch of the mess in neon_local